upload/helpers.py

```diff
@@ -440,19 +440,34 @@ def insert_commit(commitid, branch, pr, repository, owner, parent_commit_id=None
         },
     )
 
-    edited = False
+    edited = []
     if commit.state != "pending":
         commit.state = "pending"
-        edited = True
+        edited.append("state")
     if parent_commit_id and commit.parent_commit_id is None:
         commit.parent_commit_id = parent_commit_id
-        edited = True
+        edited.append("parent_commit_id")
     if branch and commit.branch != branch:
         # A branch head may have been moved; this allows commits to be "moved"
         commit.branch = branch
-        edited = True
+        edited.append("branch")
     if edited:
-        commit.save(update_fields=["parent_commit_id", "state", "branch"])
+        commit.save(update_fields=edited)
+
+    # Update Branch.head object for the new branch
+    # if the current commit is newer than the existing Branch.head commit
+    if "branch" in edited:
+        new_branch = Branch.objects.filter(
+            name=branch, repository_id=repository.repoid
+        ).first()
+        if new_branch:
```

Comment:
Is there a scenario where this returns `None`?

Reply:
If they specify a branch value that doesn't exist, then we don't update the Branch model